### ISSUE TITLE
How does instance segmentation annotation of large partially-occluded objects impact model training?

### ISSUE THREAD

**Issue by Abdul-Mukit** [roboflow/rf-detr#490]
Hello,
I need some advice regarding annotating large partially-occluded objects and how it fits with the loss calculation for instance segmentation in RFDETRSeg.

I have a 4-class dataset. Classes 0, 1 are usually small/medium-sized objects, e.g., cardboard boxes. Classes 2 and 3 are always very big, e.g., a large piece of paper or a table-top. The objects of classes 0 and 1 are always placed on top of objects of classes 2 and 3. This most often creates an occluded view of the large objects. This creates confusion on how the large objects should be annotated and how that would impact the model's learning to produce confident masks for those objects. 

Easy example:  
A couple of cardboard boxes are on a table, and they are completely occluding the left half of the table. Then, I only annotate the visible right half of the table. 

Harder example:
There is a cardboard box on top of a table. The box is in the middle of the table.  Following is an unannotated image. 

<img width="921" height="643" alt="Image" src="https://github.com/user-attachments/assets/db831a7c-e74f-479c-9790-325dea376a79" />  

Option 1: Annotate the table fully by disregarding the box in the middle. Annotate the box as usual. During training, the table's ground-truth 2D mask will contain the box.  
<img width="937" height="649" alt="Image" src="https://github.com/user-attachments/assets/b53b07e0-292c-4690-91e8-503ec0c14936" />  

Option 2: Annotate the table with polygons very carefully to wrap around the box. During training the 2D mask for the table will no longer contain the box. 
<img width="951" height="639" alt="Image" src="https://github.com/user-attachments/assets/1f523910-78f0-4a05-b609-264c68168b5d" />  

Which annotation style is better when it comes to the mask prediction-related loss functions used in RFDETRSeg?  Would appreciate your view regarding this when it comes to how to model's loss functions will be impacted. Will option-1 confuse the model?

There could also be mixed cases. For example, the left-half of the table is covered by boxes. There is one box at the center of the right half of the table.

I have been observing very low confidence for the large category objects. Not sure if this is due to the annotation style mistake. I have been disregarding the occluding small objects. Or maybe it is simply because of data imbalance. In an image, there can be only one large table, but many cardboard boxes on it.

**Comment by SebastianJanampa** [comment]
Hi,
In my opinion, I think option 2 is the best way to train the RF-DETR seg model. Option 1 will confuse the model because you're essentially saying that the box's pixels correspond to two different objects, which is incorrect. 

Low confidence can be caused by low-quality ground-truth masks, class imbalance, and short training. A quick solution for better mask quality is to use sam2 (or sam3) to create the masks.

**Comment by Abdul-Mukit** [comment]
@SebastianJanampa, hi again. Thank you.  

Yes, I also think option 1 might be questionable. But option 2 is expensive when it comes to annotating thousands of images in a dataset. My hope was that the model would learn that large objects can regularly be occluded and be confident about it.

I have used both SAM2 and SAM3. They are undoubtedly impressive. But they are quite a bit unreliable when the target items are not commonplace. The above examples are just dummy examples. In industry applications, we often have very challenging lighting conditions and peculiar/irregularly-shaped items to detect. Even after text prompting and prompts using points, bboxes, etc, SAM often fails to get the correct masks. Both SAM2 and 3 need significant human intervention to ensure quality annotations.
